### ISSUE TITLE
personel config lost after runnig linutil

### ISSUE THREAD

**Issue by salahoued** [ChrisTitusTech/linutil#109]
## Describe the bug
I use `stow` to manage my `dotfiles`, so my personel configs under `.config` are mostly `symbolic links` ,
So when setting up (for example) `rofi`, `linutil` tries to backup the `.config/rofi` with `.config/rofi.bak`, but it will end up as a `symlink` pointing to the same directory as `.config/rofi`, resulting in losing any old/previous configurations.

## To Reproduce
1. having a `symlink` in `~/.config` pointing to different directory: `~/.config/rofi`  -> `~/Documents/config/rofi`
2. run `linutil`
3. try `to set up` rofi
4. see that `~/.config/rofi` and `~/.config/rofi.bak` are pointing to the same `~/Documents/config/rofi` and the old `config.rasi` is lost

## Expected behavior
maybe check if `~/.config/rofi`  is a symlink or a directory before backing it up.
- either delete the symlink altogether and create a new `~/.config/rofi` **directory** where the new config will reside
- or walk through `~/.config/rofi` and backing up each and every file and directory before copying the new config

personally, I prefer the second option this way the user, can still keep his old config and enjoy the new one, while still managing it with `dotfiles` management solution.

## Screenshots
![sip](https://github.com/user-attachments/assets/a262a8be-50f8-4ae5-9d59-c6a56480dd4e)

**Comment by salahoued** [comment]
maybe something like this in `rofi-setup.sh`
```
if [ -h "$HOME/.config/rofi" ]; then
        for i in "$HOME/.config/rofi"/*; do
            cp -r cp -r "$HOME/.config/rofi/$i" "$HOME/.config/rofi/$i.bak"
        done
    elif [ ! -h "$HOME/.config/rofi" -a -d "$HOME/.config/rofi" ]; then
        cp -r "$HOME/.config/rofi" "$HOME/.config/rofi.bak"
    fi
```
or add a `checkIsSymlink()` in `common-script.sh` that will be used in any dotfiles set up

**Comment by lj3954** [comment]
Can you try adding the -L flag to the cp command? That option should first dereference all symlinks.

**Comment by salahoued** [comment]
Yes, that works, there is now an actual directory  not symlink `~/.config/rofu.bak` with the all the files in it,

This is just a personal opinion, but it would be better to have it this way: for example move `~/.config/rasi/config.rasi` to `~/.config/rofi/config.rasi.bak` and any other files in it be renamed in the same way if it exists in the old config.

thanks
